### PR TITLE
Add labels to references

### PR DESCRIPTION
**Pull request by discodavey**

### CODE REVIEW

File: source/_patterns/00-atoms/references/reference.yaml [hunk 1]
```diff
@@ -4,7 +4,10 @@ properties:
   bibId:
     type: string
     minLength: 1
-  titleLink:
+  title:
```
Comment (nlisgo) [inline source/_patterns/00-atoms/references/reference.yaml:7]: This has changed to `title`, should be `titleLink`?

File: source/_patterns/00-atoms/references/reference.mustache [hunk 2]
```diff
@@ -1,5 +1,7 @@
 <div class="reference"{{#bibId}} data-popup-label="See in references" data-popup-contents="{{bibId}}" id="{{bibId}}"{{/bibId}}>
-
+  {{#hasReferenceLabel}}
```
Comment (nlisgo) [inline source/_patterns/00-atoms/references/reference.mustache:2]: This should be `{{#referenceLabel}}`

File: source/_patterns/00-atoms/references/reference.mustache [hunk 3]
```diff
@@ -39,11 +41,23 @@
     {{> atoms-doi}}
   {{/doi}}
 
+  {{^doi}}
```
Comment (nlisgo) [inline source/_patterns/00-atoms/references/reference.mustache:44]: this has snuck in from https://github.com/elifesciences/pattern-library/pull/1063, I think

File: source/_patterns/00-atoms/references/reference.mustache [hunk 4]
```diff
@@ -39,11 +41,23 @@
...
+    {{/titleLink}}
+
+  {{/doi}}
+
   {{#hasAbstracts}}
     <ul class="reference__abstracts">
   {{/hasAbstracts}}
   {{#abstracts}}
-    <li class="reference__abstract"><a href="{{url}}" class="reference__abstract_link">{{{name}}}</a></li>
+    <li class="reference__abstract">
```
Comment (nlisgo) [inline source/_patterns/00-atoms/references/reference.mustache:58]: Undo this change

Comment (discodavey) [reply source/_patterns/00-atoms/references/reference.mustache:58]: Can you explain why?

File: source/_patterns/00-atoms/references/reference.yaml (hunk 1 above)
Comment (discodavey) [reply source/_patterns/00-atoms/references/reference.yaml:7]: Yes this should be titleLink. Not sure why it has been changed back

File: source/_patterns/00-atoms/references/reference.mustache (hunk 3 above)
Comment (discodavey) [reply source/_patterns/00-atoms/references/reference.mustache:44]: I had to put this in so I could style this up. Can be removed when we merge in to master. If there is another reason can you let me know.

File: source/_patterns/00-atoms/references/reference.mustache (hunk 4 above)
Comment (discodavey) [reply source/_patterns/00-atoms/references/reference.mustache:58]: Can you explain why as I have just moved the link to the line below as just tidying up the code.

File: source/_patterns/00-atoms/references/reference.mustache [hunk 5]
```diff
@@ -1,5 +1,7 @@
 <div class="reference"{{#bibId}} data-popup-label="See in references" data-popup-contents="{{bibId}}" id="{{bibId}}"{{/bibId}}>
-
+  {{#referenceLabel}}
```
Comment (nlisgo) [inline source/_patterns/00-atoms/references/reference.mustache:2]: Do you have any objection to just calling this `label` and dropping the prefix?

Comment (nlisgo) [reply source/_patterns/00-atoms/references/reference.mustache:2]: Or `type`?

Comment (discodavey) [reply source/_patterns/00-atoms/references/reference.mustache:2]: I have replied to this in the main ticket. https://app.zenhub.com/workspaces/journal-5ede53e8b9b11773fd13f981/issues/elifesciences/issues/6468
